numpy/lib/histograms.py

```diff
@@ -349,6 +378,44 @@ def _unsigned_subtract(a, b):
         # signed to unsigned
         return np.subtract(a, b, casting='unsafe', dtype=dt)
 
+def _get_auto_bin_edges(a, n_equal_bins, first_edge, last_edge):
+    """
+    Computes the bins used internally by `np.histogram(bins='auto')`.
+
+    Parameters
+    ==========
+    a : ndarray
+        Ravelled data array
+    first_edge, last_edge
+        Forwarded arguments from `_get_outer_edges`.
+    n_equal_bins : 
+        Forwarded arugment from _get_bin_edges,
+        this function only called when n_equal_bins > 2
+
+    Returns
+    =======
+    bin_edges : ndarray
+        Array of bin edges
+    """
+    keep = (a >= first_edge)
+    keep &= (a <= last_edge)
+    if not np.logical_and.reduce(keep):
+        a = a[keep]
+
+    if a.size == 0:
+        return np.array([0,1])
+
+    width = np.ptp((first_edge,last_edge))/(n_equal_bins)
```

Comment:
`ptp` is overkill here for subtracting two scalars, one of which you know to be larger than the other.
```suggestion
    width = _unsigned_subtract(last_edge, first_edge) / n_equal_bins
```